src/components/viewContainer.js

```diff
@@ -1,115 +1,106 @@
-define(['browser', 'dom', 'layoutManager', 'css!components/viewManager/viewContainer'], function (browser, dom, layoutManager) {
-    'use strict';
+import 'css!components/viewManager/viewContainer';
+/* eslint-disable indent */
 
     function setControllerClass(view, options) {
         if (options.controllerFactory) {
             return Promise.resolve();
         }
 
-        var controllerUrl = view.getAttribute('data-controller');
+        let controllerUrl = view.getAttribute('data-controller');
 
         if (controllerUrl) {
             if (controllerUrl.indexOf('__plugin/') === 0) {
                 controllerUrl = controllerUrl.substring('__plugin/'.length);
             }
 
             controllerUrl = Dashboard.getConfigurationResourceUrl(controllerUrl);
-            return getRequirePromise([controllerUrl]).then(function (ControllerFactory) {
+            return import(controllerUrl).then((ControllerFactory) => {
```

Comment:
It's importing random stuff, so I'm not too sure.

It looks like it's used for frontend plugin controllers (Web plugins can add pages to the client), but we don't have any to test it, so...

Reply:
I don't know how relevant [this](https://github.com/jellyfin/jellyfin-web/blob/master/src/controllers/music/musicrecommended.js#L304) is, but for moviesreccomended/musicrecommended/tvrecommended, I have used `default` for the import. 

From what I have read, if a module is imported as above (`import(X).then((Y) => {`), when `Y` is called, it will need to be called as `Y.default(...`.
If it is imported as `import(X).then(({default: Y}) => {`, then it should be called as `Y(...`

This is also making the assumption that all `X` have a default export.

Hopefully that made some amount of sense.

Reply:
Looking at the source, we never set the `data-controller` attribute, so this code is never reached, I think?

Reply:
I don't think so either. So I guess it's more unused code.